### PR TITLE
state: adds offset access and manipulation

### LINKED ISSUE #495: manipulating ListState/TableState offset
## Problem

The `offset` of `ListState` and `TableState` can neither be accessed nor manipulated. 

## Solution

Add publicly visible `fn offset` and `fn offset_mut` to both `ListState` and `TableState`. 

## Alternatives

None.

## Additional context

Offset manipulation is needed to implement controls to center the view around the currently selected list item or row, e.g. like Emacs' `recenter-top-bottom` via `C-l` shortcut.

### PR DESCRIPTION
**Pull request by wookietreiber**
## Description

Adds offset access and manipulation. Fixes #495. 

## Testing guidelines

I have a few structs that keep the table state. I use this `trait ScrollableTable` for each of these to attach actions like scroll up/down, etc. I've added this `fn recenter` to test offset manipulation. `ListState` should be pretty much analogously. 

Note that I haven't added tests per se, because I think that these kinds of accessor/mutator functions are quite obvious. 

```rust
use tui::layout::Rect;
use tui::widgets::TableState;

pub trait ScrollableTable {
    fn size(&self) -> Rect;
    fn state(&self) -> &TableState;
    fn state_mut(&mut self) -> &mut TableState;

    fn recenter(&mut self) {
        if let Some(selected) = self.state().selected() {
            let height = self.size().height as usize;

            let offset = self.state_mut().offset_mut();
            *offset = selected.saturating_sub(height / 2);
        }
    }
}
```

## Checklist

* [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
* [ ] I have added relevant tests.
* [x] I have documented all new additions.

### CODE REVIEW

File: src/widgets/list.rs [hunk 1]
```diff
@@ -23,6 +23,16 @@ impl Default for ListState {
 }
 
 impl ListState {
+    /// Returns the offset.
```
Comment (Ranguna) [inline src/widgets/list.rs:26]: Do we need these comments ?

Asking because the rest of the code doesn't seem to use comments like these.

Comment (wookietreiber) [reply src/widgets/list.rs:26]: Even if there aren't many doc comments as of now, why not start?

Comment (Ranguna) [reply src/widgets/list.rs:26]: I personally prefer comments only when strictly needed, the code you wrote looks self explanatory to me, so we don't need to maintain code + comments.

But I'll leave this to the maintainer to review 👌

**Comment by wookietreiber** [comment]
cargo make does not seem to be compatible with rustc 1.56.1: 

```
error: failed to compile `cargo-make v0.35.8`, intermediate artifacts can be found at `/tmp/cargo-install52QPLl`

Caused by:
  package `os_str_bytes v6.2.0` cannot be built because it requires rustc 1.57.0 or newer, while the currently active rustc version is 1.56.1
Error: Process completed with exit code 101.
```

**Pull request closed without merging**